### DISCUSSION TITLE
Termux Api Commands

### DISCUSSION THREAD

**Discussion by sleeping3119** [termux/termux-app#3512] [Q&A] [answered]
is there anyway to run termux-api commands after using su, no matter if i switch user to termux itself, no termux-api command work. Ofcourse i added path of bin and did not change the shell. 
I tried:
`su -ps $(SHELL) `
`tsu `
i also tried `env > bash.txt` before switching user and 
`source bash` after.
Please tell me on which factor termux api command depends, what get missed when i run su with same shell and environment variables.

**Comment by sylirre** [comment, marked as answer]
The problem with Termux:API is that its command line frontend tool (`termux-api`, usable by all API wrapper scripts) creates socket files which must be accessible by Termux:API application.

As you are doing so under `su`, the sockets will have ownership and SELinux labels preventing Termux:API from accessing them.
